src/promptflow/promptflow/executor/_line_execution_process_pool.py

```diff
@@ -152,52 +91,82 @@ def __init__(
-            multiprocessing_start_method = None
-        self.context = get_multiprocessing_context(multiprocessing_start_method)
-        use_fork = self.context.get_start_method() == "fork"
-        # When using fork, we use this method to create the executor to avoid reloading the flow
-        # which will introduce a lot more memory.
-        if use_fork:
-            self._executor_creation_func = partial(create_executor_fork, flow_executor=flow_executor)
-        elif flow_executor._flow_file:
-            self._executor_creation_func = partial(
-                FlowExecutor.create,
-                flow_file=flow_executor._flow_file,
-                connections=flow_executor._connections,
-                working_dir=flow_executor._working_dir,
-                raise_ex=False,
-            )
-        else:  # Legacy flow executor, will be deprecated with the legacy pf portal.
-            self._executor_creation_func = partial(
-                create_executor_legacy,
-                flow=flow_executor._flow,
-                connections=flow_executor._connections,
-                loaded_tools=flow_executor._loaded_tools,
-                cache_manager=flow_executor._cache_manager,
-            )
+        use_fork = "fork" in sys_start_methods
+        self.context = get_multiprocessing_context("fork" if use_fork else "spawn")
+        self._flow_file = flow_executor._flow_file
+        self._connections = flow_executor._connections
+        self._working_dir = flow_executor._working_dir
+
+        if not use_fork:
+            if flow_executor._flow_file:
+                self._executor_creation_func = partial(
+                    FlowExecutor.create,
+                    flow_file=self._flow_file,
+                    connections=self._connections,
+                    working_dir=self._working_dir,
+                    raise_ex=False,
+                )
+            else:  # Legacy flow executor, will be deprecated with the legacy pf portal.
+                self._executor_creation_func = partial(
+                    create_executor_legacy,
+                    flow=flow_executor._flow,
+                    connections=self._connections,
+                    loaded_tools=flow_executor._loaded_tools,
+                    cache_manager=flow_executor._cache_manager,
+                )
         self._use_fork = use_fork
         self._storage = flow_executor._run_tracker._storage
         self._flow_id = flow_executor._flow_id
         self._log_interval = flow_executor._log_interval
-        self._line_timeout_sec = flow_executor._line_timeout_sec
+        self._line_timeout_sec = get_int_env_var("PF_LINE_TIMEOUT_SEC", flow_executor._line_timeout_sec)
         self._output_dir = output_dir
 
     def __enter__(self):
         manager = Manager()
         self._processing_idx = manager.dict()
         self._completed_idx = manager.dict()
 
-        self._inputs_queue = Queue()
+        self._task_queue = Queue()
```

Comment:
Why are we using Queue here and manager.Queue in line 135?

Reply:
fixed

Reply:
Queue() uses a synchronization primitive called a “lock” to ensure that only one process can access
 the queue at a time
 While manager().Queue() uses a manager object to create a queue that can be shared
between multiple processes.
Use Queue() will case an error: "A SemLock created in a fork context is being shared with a process
in a spawn context. This is not supported".
 So use the manager.Queue().